packages/next-codemod/lib/parser.ts

```diff
@@ -0,0 +1,24 @@
+import j from 'jscodeshift'
+import babylonParse from 'jscodeshift/parser/babylon'
+import tsOptions from 'jscodeshift/parser/tsOptions'
+
+const dtsOptions = {
+  ...tsOptions,
+  plugins: [
+    ...tsOptions.plugins.filter((plugin) => plugin !== 'typescript'),
+    ['typescript', { dts: true }],
+  ],
+}
+
+function createParserFromPath(filePath: string): j.JSCodeshift {
+  const isTypeFile = /\.d\.(m|c)?ts$/.test(filePath)
+  if (isTypeFile) {
+    return j.withParser(babylonParse(dtsOptions))
+  }
+  // In Next.js js files could also contain jsx syntax, applying with tsx parser is most compatible.
+  // This is similar to how Next.js itself handles tsx and ts files with SWC.
+  const isTs = filePath.endsWith('.ts')
```

Comment:
Not accounting for `.mts` or `.cts`. It's easier to flip it and check for `.tsx` instead since JSX + TS only has `.tsx` I believe.

Reply:
We cannot feed `.js` and `.jsx` files into ts parser, it will fail with jsx. I extend the condition to support mts and cts